build.gradle

```diff
@@ -1,25 +1,26 @@
 plugins {
     id "application"
     id "checkstyle"
+    id "com.github.spotbugs" version "5.0.6"
```

Comment:
Cool integration. So how do we actually use this in practice? Before launching a new release we run both checkstyleMain / Test and spotbugsMain / Test from the IntelliJ Gradle console and fix whatever pops up?

Reply:
See commit #2 - `./gradlew :check` will run all the relevant things, and as of this PR, that will be run on every PR as well.

Reply:
Oh nice, so it's run automatically via CI? Where should we keep an eye on for the output if these find anything?

I ran `check` locally using Gradle and it seems to run through all these. However, for `checkstyleMain` I'm now getting:
`warning: [options] module name in --add-exports option not found: org.junit.platform.engine`

And for `spotbugsTest` and `spotbugsMain` I'm getting:
```
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by edu.umd.cs.findbugs.ba.jsr305.TypeQualifierValue (file:/C:/Users/...n/.gradle/caches/modules-2/files-2.1/com.github.spotbugs/spotbugs/4.6.0/9b88f95eba7b8de50339cd844e251bb7c6eca68a/spotbugs-4.6.0.jar)
WARNING: Please consider reporting this to the maintainers of edu.umd.cs.findbugs.ba.jsr305.TypeQualifierValue
WARNING: System::setSecurityManager will be removed in a future release
```

Should I file a new bug for these, or easy to resolve via this PR?

Reply:
That's [this bug](https://github.com/spotbugs/spotbugs/issues/1579).